cloudcix_primitives/vlanifns.py

```diff
@@ -0,0 +1,245 @@
+            type: string
+            required: true
+        namespace:
+            description: The VRF network namespace identifier, such as 'VRF123'.
+            type: string
+            required: true
+
+    return:
+        description: |
+            A tuple with a boolean flag indicating if the VLAN tagged interface creation was successful,
+            and the output or error message.
+        type: tuple
+    """
+    # Define message
+    messages = {
+        1000: f'Successfully created interface {ifname}.{vlan} inside namespace {namespace} ',
+        1001: f'Interface {ifname}.{vlan} already exists inside namespace {namespace} ',
+
+        3021: f'Failed to connect to the enabled PodNet for vlanif_check payload:  ',
+        3022: f'Failed to connect to the enabled PodNet for vlanif_add payload:  ',
+        3023: f'Failed to run vlanif_add payload on the enabled PodNet. Payload exited with status ',
+        3024: f'Failed to connect to the enabled PodNet for vlanif_ns payload:  ',
+        3025: f'Failed to run vlanif_ns payload on the enabled PodNet. Payload exited with status ',
+        3026: f'Failed to connect to the enabled PodNet for vlanif_up payload:  ',
+        3027: f'Failed to run vlanif_up payload on the enabled PodNet. Payload exited with status ',
+
+        3051: f'Failed to connect to the disabled PodNet for vlanif_check payload:  ',
+        3052: f'Failed to connect to the disabled PodNet for vlanif_add payload:  ',
+        3053: f'Failed to run vlanif_add payload on the disabled PodNet. Payload exited with status ',
+        3054: f'Failed to connect to the disabled PodNet for vlanif_ns payload:  ',
+        3055: f'Failed to run vlanif_ns payload on the disabled PodNet. Payload exited with status ',
+        3056: f'Failed to connect to the disabled PodNet for vlanif_up payload:  ',
+        3057: f'Failed to run vlanif_up payload on the disabled PodNet. Payload exited with status ',
+    }
+
+    # Default config_file if it is None
+    if config_file is None:
+        config_file = '/opt/robot/config.json'
+
+    status, config_data, msg = load_pod_config(config_file)
+    if not status:
+      if config_data['raw'] is None:
+          return False, msg
+      else:
+          return False, msg + "\nJSON dump of raw configuration:\n" + json.dumps(config_data['raw'],
+              indent=2,
+              sort_keys=True)
+    enabled = config_data['processed']['enabled']
+    disabled = config_data['processed']['disabled']
+
+
+    def run_podnet(podnet_node, prefix, successful_payloads):
+        rcc = SSHCommsWrapper(comms_ssh, podnet_node, 'robot')
+        fmt = PodnetErrorFormatter(
+            config_file,
+            podnet_node,
+            podnet_node == enabled,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads
+        )
+
+        payloads = {
+            'vlanif_check' : f'ip netns exec {namespace} ip link show {ifname}.{vlan}',
```

Comment:
@oscar-lenin  If the VLAN interface is linked the namespace, does this mean that is up?

Is it possible for it to be linked and down? If yes, to ensure it is up, should the `vlanif_up` payload always be executed?

For example, in the [ns primitive](https://github.com/CloudCIX/primitives/blob/d60c73c4c38b1341cfe2024ff30e1fbb09e77fcc/cloudcix_primitives/ns.py#L127), the `create_namespace` is only executed if the payload for `find_namespace` is unsuccessful. However, the remaining payloads are always executed to ensure it is in the expected state at the end of the build request.

Reply:
Hi Maria, that makes sense. I've changed it so vlanif_up is always run even if the interface already exists.